azext_edge/edge/providers/support/base.py

```diff
@@ -386,34 +353,33 @@ def process_cron_jobs(
     )
 
     return _process_kubernetes_resources(
-        sub_group=sub_group,
+        directory_path=directory_path,
         resources=cron_jobs,
-        resource_api=resource_api,
         prefix_names=prefix_names,
         kind=BundleResourceKind.cronjob.value,
     )
 
 
 def assemble_crd_work(
     apis: Iterable[EdgeResourceApi],
-    sub_group: Optional[str] = None,
     file_prefix_map: Optional[Dict[str, str]] = None,
+    sub_group: Optional[str] = None,
```

Comment:
Suggestion...instead of having `sub_group` parameter here for `assemble_crd_work()` (it is the only place after this change set right?) we rename it to `directory_path: Optional[str] = None` whose intent is to override the default directory path which is the api.monikor.

The current directory_path logic on line 374 should move before the second loop (currently line 373) because the api is the same when iterating through its kinds, and can be refactored to:

```python
if not directory_path:
    directory_path = api.moniker
```

Then usage of `directory_path` override, such as in billing line 109 can do

```python
billing_to_run.update(assemble_crd_work(apis=apis, directory_path=ARC_BILLING_DIRECTORY_PATH))
```